pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -744,17 +746,21 @@ func buildGatewayNetworkFiltersFromTCPRoutes(node *model.Proxy, push *model.Push
 			continue
 		}
 
+		if server.GetTls().GetMode() == networking.ServerTLSSettings_ISTIO_MUTUAL {
+			out = append(out, xdsfilters.TCPListenerMx)
+		}
+
 		// ensure we satisfy the rule's l4 match conditions, if any exist
 		// For the moment, there can be only one match that succeeds
 		// based on the match port/server port and the gateway name
 		for _, tcp := range vsvc.Tcp {
 			if l4MultiMatch(tcp.Match, server, gateway) {
-				return buildOutboundNetworkFilters(node, tcp.Route, push, port, v.Meta)
+				out = append(out, buildOutboundNetworkFilters(node, tcp.Route, push, port, v.Meta)...)
```

Comment:
What is `vsvc.Tcp` has multiple elements? or multiple virtualServices.. won't we get duplicates?